edge/pkg/edgehub/common/http/http_test.go

```diff
@@ -258,6 +258,10 @@ func TestBuildRequest(t *testing.T) {
 				t.Errorf("BuildRequest() error = %v, expectedError = %v", err, tt.wantErr)
 				return
 			}
+			if got == nil {
+				// will never execute, but can explicitly prevent nil warnings when checking
+				return
+			}
```

Comment:
seem that `BuildRequest` method will not return nil `got` when `err != nil`

Reply:
Yes, I looked at the code for the BuildRequest and was sure that the added code was never executed, but doing so explicitly organizes editing tools and lint's nil warnings.

I can remove this part of the code if needed

Reply:
In theory, if it is not for special circumstances, according to the specification, when `err == nil`, the returned value must be a valid value.

Reply:
Okay, I've removed that part of the code